### PR TITLE
feat: batch events & improve logging

### PR DESCRIPTION
**Pull request by paolodamico**
- Batches event reporting to Braze using `exportEvents` to improve plugin engine performance.
- Better debugging by adding event ID details.
@hazzadous

### CODE REVIEW

File: index.ts [hunk 1]
```diff
@@ -878,50 +887,64 @@ const _handleOnEvent = async (pluginEvent: PluginEvent, meta: BrazeMeta): Promis
...
-    // See https://www.braze.com/docs/api/endpoints/user_data/post_user_track/
-    // for more info.
+export const exportEvents = async (pluginEvents: PluginEvent[], meta: BrazeMeta): Promise<void> => {
+    // NOTE: batching events in 75 events per request
+    const batchSize = 75
+    const batches: PluginEvent[][] = []
 
-    if (!meta.config.eventsToExport && !meta.config.userPropertiesToExport) {
-        return
+    for (let i = 0; i < pluginEvents.length; i += batchSize) {
```
[the remark this answers is not in the document]
Comment (igorosip0v) [reply index.ts]: @paolodamico I've refactored the way we batching.

File: tests/export-events-network-error.test.ts [hunk 2]
```diff
@@ -1,7 +1,7 @@
 import { RetryError } from '@posthog/plugin-scaffold'
 import fetch from 'node-fetch'
 
-import { BrazeMeta, onEvent, setupPlugin } from '..'
+import { BrazeMeta, exportEvents, setupPlugin } from '..'
 
 jest.mock('node-fetch')
```
Comment (hazzadous) [inline tests/export-events-network-error.test.ts:6]: Off topic:

As you didn't change this, but one big advantage of using something like MSW over `node-fetch` is we can be pretty sure that if we mock out with a timeout or 5xx error then the error propagation will be representative of non-mocked usage.

**Review by hazzadous — APPROVED** [review]
Looks good. The one thing I'd add is a test verifying functionality of sending multiple events to `exportEvents`

File: index.ts [hunk 3]
```diff
@@ -84,17 +85,20 @@ export async function setupPlugin({ config, global }: BrazeMeta): Promise<void>
                 timeout: 5000,
             })
         } catch (e) {
-            console.error(e)
+            console.error(e, endpoint, options.body, eventUuid)
             throw new RetryError('Fetch failed, retrying.')
```
Comment (hazzadous) [inline index.ts:89]: Ah also, additionally, we should not be retrying e.g. 400 errors, something I missed previously!

Essentially we only want to retry on intermittent errors e.g. 5xx or not response.

Comment (hazzadous) [reply index.ts:89]: For example, if we have a payload that is too large and get 413 entity too large, we will continuously churn on these.

Comment (hazzadous) [reply index.ts:89]: Looks like that is handled further down as well, so I think we just want to remove the `throw` 🤔

**Pull request merged**